src/com/google/cose/Ec2SigningKey.java

```diff
@@ -114,6 +118,74 @@ public ECPublicKey getPublicKey() {
     return (ECPublicKey) this.keyPair.getPublic();
   }
 
+  private static byte[] arrayFromBigNum(BigInteger n, int curveSize) {
+    byte[] rgb = new byte[(curveSize + 7) / 8];
+    byte[] rgb2 = n.toByteArray();
+    if (rgb.length == rgb2.length) {
+      return rgb2;
+    }
+    if (rgb2.length > rgb.length) {
+      System.arraycopy(rgb2, rgb2.length - rgb.length, rgb, 0, rgb.length);
+    } else {
+      System.arraycopy(rgb2, 0, rgb, rgb.length - rgb2.length, rgb2.length);
+    }
+    return rgb;
+  }
+
+  /** Generates a COSE formatted Ec2 signing key given a specific algorithm */
+  public static Ec2SigningKey generateKey(Algorithm algorithm) throws CborException, CoseException {
+    KeyPair keyPair;
+    int curveSize;
+    int header;
+    String curveName;
+
+    switch (algorithm) {
+      case SIGNING_ALGORITHM_ECDSA_SHA_256:
+        curveName = "secp256r1";
+        curveSize = 256;
```

Comment:
If you make this `curveByteSize` instead, then there's no need for the round-up arithmetic elsewhere.

Just a suggestion.

Reply:
I agree that a variable name like `curveByteSize` would clue people onto why `32` would be paired with `secp256r1`, but I think it would be clearer to leave it at bits. I added a comment indicating the roundup arithmetic.